c++/cpp2py/converters/tuple.hpp

```diff
@@ -11,28 +11,28 @@ namespace cpp2py {
     using tuple_t = std::tuple<Types...>;
 
     // c2py implementation
-    template <std::size_t... Is> static PyObject *c2py_impl(tuple_t const &t, std::index_sequence<Is...>) {
-      auto objs        = std::array<pyref, sizeof...(Is)>{pyref(py_converter<Types>::c2py(std::get<Is>(t)))...};
+    template <std::size_t... Is> static PyObject *c2py_impl(tuple_t t, std::index_sequence<Is...>) {
+      auto objs        = std::array<pyref, sizeof...(Is)>{pyref(py_converter<std::decay_t<Types>>::c2py(std::move(std::get<Is>(t))))...};
       bool one_is_null = std::accumulate(std::begin(objs), std::end(objs), false, [](bool x, PyObject *a) { return x or (a == NULL); });
       if (one_is_null) return NULL;
       return PyTuple_Pack(sizeof...(Types), (PyObject *)(objs[Is])...);
     }
 
     // is_convertible implementation
     template <int N, typename T, typename... Tail> static bool is_convertible_impl(PyObject *seq, bool raise_exception) {
```

Comment:
We should use a index_seq also here. These multiple recursive calls are old-fashioned, and a lot more work to compile...

Reply:
a simple fold with `and ...` will also short circuit

Reply:
Agreed! Changed accordingly.